src/App.tsx

```diff
@@ -15,48 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Name = 'name',
+  Length = 'length',
+  Default = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortBy,
+  isReversed: boolean,
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortBy.Name: (
+        prepearedGoods.sort((goods1, goods2) => (
+          goods1.localeCompare(goods2)
+        ))
+      );
+        break;
+
+      case SortBy.Length: (
+        prepearedGoods.sort((goods1, goods2) => (
+          goods1.length - goods2.length
+        ))
+      );
+        break;
+
+      default:
+        return prepearedGoods;
+    }
+  }
+
+  if (isReversed) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortBy.Default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    isReversed,
+  );
+
+  const isSortFieldReversed = sortField || isReversed;
+
+  const handleResetClick = () => {
+    setSortField(SortBy.Default);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SortBy.Name })}
```

Comment:
```suggestion
          className={cn(
            'button',
            'is-info',
            { 'is-light': sortField !== SortBy.Name },
          )}
```
Write every class in className function parameters separately, also pay attention to your code style. Fix in all places